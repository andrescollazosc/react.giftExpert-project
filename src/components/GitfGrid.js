import React from "react";
import { useFetchGifts } from "../hooks/useFetchGifts";
// import { getGift } from "../helpers/getGifts";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GitfGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data, loading } = useFetchGifts(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading ?? <p className="animate__animated animate__fadeIn">Loading</p>}

      <div className="card-grid">
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GitfGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
