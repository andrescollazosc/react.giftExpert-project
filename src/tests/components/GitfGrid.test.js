import { shallow } from "enzyme";
import { GitfGrid } from "../../components/GitfGrid";
import React from "react";
import "@testing-library/jest-dom";
import { useFetchGifts } from "../../hooks/useFetchGifts";
jest.mock("../../hooks/useFetchGifts");

describe("Testing <GitfGrid />", () => {
  const category = "Dragon";

  test("should be catch the snapshot", () => {
    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GitfGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should be show item when images is loaded", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://mundoloco/imagen.gif",
        title: "Cualquier cosa",
      },
      {
        id: "DFC",
        url: "https://mundoloco/imagen_1.gif",
        title: "Otra cosa Â¿",
      },
    ];

    useFetchGifts.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GitfGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);

    console.log(wrapper.find("GifGridItem").length);

    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
