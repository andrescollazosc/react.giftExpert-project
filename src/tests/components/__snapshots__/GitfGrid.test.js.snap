// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GitfGrid /> should be catch the snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon
  </h3>
  <Component />
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing <GitfGrid /> should be show item when images is loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://mundoloco/imagen.gif"
    />
    <GifGridItem
      id="DFC"
      key="DFC"
      title="Otra cosa Â¿"
      url="https://mundoloco/imagen_1.gif"
    />
  </div>
</Fragment>
`;
